OpenRA.Mods.Common/Lint/CheckActors.cs

```diff
@@ -18,10 +19,16 @@ public class CheckActors : ILintMapPass
 	{
 		public void Run(Action<string> emitError, Action<string> emitWarning, ModData modData, Map map)
 		{
-			var actorTypes = map.ActorDefinitions.Select(a => a.Value.Value);
-			foreach (var actor in actorTypes)
-				if (!map.Rules.Actors.Keys.Contains(actor.ToLowerInvariant()))
-					emitError($"Actor `{actor}` is not defined by any rule.");
+			var scriptBindings = Game.ModData.ObjectCreator.GetTypesImplementing<ScriptGlobal>().SelectMany(t => Utility.GetCustomAttributes<ScriptGlobalAttribute>(t, true)).ToArray();
```

Comment:
Consider `.Select(x => x.Name).ToHashSet()` for fast `Contains` check inside the loop. (I'm not sure if there's many bindings but just in case)